src/test/java/com/google/gcp_variant_transforms/library/VariantToBqUtilsTest.java

```diff
@@ -0,0 +1,248 @@
+   * If genotypes in missing, genotypes should be default value -1
+   * If the phaseSet(PS) is not presented, phase set should be default "*"
+   * Test Genotype:
+   * FORMAT           sample
+   * GT:DP:PS:HQ    1|2:1:0:23,27
+   * GT:DP:HQ         .:1:.,.
+   * In the second record, genotypes will set [-1,-1] and phase set will set "*",
+   * and HQ will be [null, null]
+   */
+  @Test
+  public void testAddCalls_whenCheckingGenotypeElements_thenTrue() {
+    Genotype sample = mock(Genotype.class);
+    when(sample.getSampleName()).thenReturn("sample");
+    // mock genotypes
+    Map<String, Integer> alleleIndexingMap = new HashMap<>();
+    Allele firstAllele = mock(Allele.class);
+    when(firstAllele.getDisplayString()).thenReturn("G");
+    alleleIndexingMap.put("G", 1);
+    Allele secondAllele = mock(Allele.class);
+    when(secondAllele.getDisplayString()).thenReturn("A");
+    alleleIndexingMap.put("A", 2);
+    when(sample.getAlleles()).thenReturn(Arrays.asList(firstAllele, secondAllele));
+
+    // mock info
+    when(sample.hasAD()).thenReturn(false);
+    when(sample.hasDP()).thenReturn(true);
+    when(sample.getDP()).thenReturn(1);
+    when(sample.hasGQ()).thenReturn(false);
+    when(sample.hasPL()).thenReturn(false);
+    Map<String, Object> extendedAttributes = new HashMap<>();
+    extendedAttributes.put("HQ", "23,27");
+    extendedAttributes.put("PS", "0");
+    when(sample.getExtendedAttributes()).thenReturn(extendedAttributes);
+
+    // mock VCF header
+    VCFFormatHeaderLine DPMetadata = mock(VCFFormatHeaderLine.class);
+    when(vcfHeader.getFormatHeaderLine("DP")).thenReturn(DPMetadata);
+    when(DPMetadata.getType()).thenReturn(VCFHeaderLineType.Integer);
+    when(DPMetadata.getCountType()).thenReturn(VCFHeaderLineCount.INTEGER);
+
+    VCFFormatHeaderLine HQMetadata = mock(VCFFormatHeaderLine.class);
+    when(vcfHeader.getFormatHeaderLine("HQ")).thenReturn(HQMetadata);
+    when(HQMetadata.getType()).thenReturn(VCFHeaderLineType.Integer);
+    when(HQMetadata.getCountType()).thenReturn(VCFHeaderLineCount.INTEGER);
+
+    // add sample to genotypeContext
+    Iterator<Genotype> genotypeIterator = mock(Iterator.class);
+    when(genotypeIterator.hasNext()).thenReturn(true, false);
+    when(genotypeIterator.next()).thenReturn(sample);
+    when(genotypesContext.iterator()).thenReturn(genotypeIterator);
+
+    // test table row fields
+    List<TableRow> calls = VariantToBqUtils.addCalls(genotypesContext, vcfHeader,
+            alleleIndexingMap);
+    TableRow row = calls.get(0);
+    assertThat(row.get(CALLS_NAME)).isEqualTo("sample");
+    assertThat(row.get(CALLS_GENOTYPE)).isEqualTo(Arrays.asList(1, 2));
+    assertThat(row.get(CALLS_PHASESET)).isEqualTo("0");
+    // PS should not be in the info map, should be set in phase set
+    assertThat(!row.containsKey("PS")).isTrue();
+    assertThat(row.get("HQ")).isEqualTo(Arrays.asList(23,27));
+
+    // Second record: contains empty fields
```

Comment:
Can we break this test into multiple tests as well?
Maybe a test function for asserting the full table row fields and another one for testing the record containing empty fields?